engine/src/build/mac/apple_sdk.py

```diff
@@ -0,0 +1,171 @@
+# Copyright 2013 The Flutter Authors. All rights reserved.
+# Use of this source code is governed by a BSD-style license that can be
+# found in the LICENSE file.
+
+import argparse
+import os
+import shutil
+import subprocess
+import sys
+
+from dataclasses import dataclass
```

Comment:
I'm curious about using this. I'd asked @jason-simmons which version of python was guaranteed to be supported across all CI platforms. He said it was 3.x but not sure about the version. Are we sure we can use dataclasses?

Reply:
dataclass was added in python 3.7, so pretty old -- 2018. I could replace it with a manually declared class but this worked for my on vpython from depot_tools, which pulls it down from cipd.

Reply:
Discussed with @jason-simmons and we believe this will work in all the environments we care about given it's been around as long as it has, and it's been verified on all our bots.